gobblin-core/src/main/java/gobblin/writer/objectstore/ObjectStoreWriter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer.objectstore;
+
+import java.io.IOException;
+
+import lombok.Getter;
+
+import com.codahale.metrics.Counter;
+
+import gobblin.writer.DataWriter;
+
+/**
+ * A writer to execute operations on a object in any object store. The record type of this writer is an {@link ObjectStoreOperation}.
+ * The {@link ObjectStoreOperation} encapsulates operation specific metadata and actions.
+ */
+@SuppressWarnings("rawtypes")
+public class ObjectStoreWriter implements DataWriter<ObjectStoreOperation> {
+
+  private final Counter operationsExecuted;
+  @Getter
+  private final ObjectStoreClient objectStoreClient;
+
+  public ObjectStoreWriter(ObjectStoreClient client) {
+    this.objectStoreClient = client;
+    this.operationsExecuted = new Counter();
```

Comment:
`operationsExecuted` should be registered with the `MetricContext`. Have `ObjectStoreWriter` extend `InstrumentedDataWriter`, then you can do `getMetricContext().counter("counter.name")`.